internal/transport/http2_client.go

```diff
@@ -951,6 +954,9 @@ func (t *http2Client) Close(err error) {
 		t.mu.Unlock()
 		return
 	}
+	if logger.V(logLevel) {
+		logger.Infof("Closing transport, will close the connection. Err: %v", err)
```

Comment:
This would end up as something like:
`Closing transport, will close the connection. Err: finished serving streams for the server transport.`

Isn't this enough:
`Closing transport: finished serving streams for the server transport`